src/vm/i386/umthunkstub.S

```diff
@@ -173,25 +173,20 @@ LOCAL_LABEL(UMThunkStub_CopyStackArgs):
 LOCAL_LABEL(InitCopyStack):
     push    ecx
     push    edx
-    mov     eax, 0  // Offset in edi
-    mov     ebx, 0  // Offset in esi
-LOCAL_LABEL(CopyStack):
     mov     edx, dword ptr [ebp - UMThunkStub_UMENTRYTHUNK_OFFSET]
     mov     edx, dword ptr [edx + UMEntryThunk__m_pUMThunkMarshInfo]
-    mov     ecx, dword ptr [edx + UMThunkMarshInfo__m_ecxArgOffset]
-    cmp     ecx, ebx
+    mov     ebx, [edx + UMThunkMarshInfo__m_cbActualArgSize]
```

Comment:
> I would expect that the m_ecxArgOffset would be set to zero, right?

I set `m_ecxArgOffset` and `m_edxArgOffset` as -1 if we don't use these registers. At the FunctionPtrTest we use stack only, so `cmp     ebx, dword ptr [edx + UMThunkMarshInfo__m_ecxArgOffset]` sets CF=1, and  `jnc     LOCAL_LABEL(CopyStack)` exit at the first iteration.